src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -2027,6 +2027,23 @@ public void visitCtTypeParameterReference(CtTypeParameterReference ref) {
 		}
 	}
 
+	private boolean printQualified(CtTypeReference<?> ref) {
+		if (importsContext.isImported(ref)) {
+			// If java.lang.Something is imported, but we are in the context of a class which is
+			// also called "Something", we should still use qualified version java.lang.Something
+			if (ref.getPackage() != null && ref.getPackage().getSimpleName().equals("java.lang")) {
```

Comment:
As an afterthought, this shouldn't probably be restricted with java.lang, but for any imported classes.